Simplenote/Classes/SPEditorTextView.m

```diff
@@ -161,17 +161,24 @@ - (BOOL)gestureRecognizer:(UIGestureRecognizer *)gestureRecognizer shouldReceive
     return [touch.view isKindOfClass:[SPTextView class]];
 }
 
-- (BOOL)becomeFirstResponder
-{
-    [self setEditable:YES];
+- (BOOL)becomeFirstResponder {
+    
+    // Editable status is true by default but we fiddle with it during setup.
+    
+    self.editable = YES;
     return [super becomeFirstResponder];
 }
 
-- (BOOL)resignFirstResponder
-{
-    BOOL response = [super resignFirstResponder];
-    [self setNeedsLayout];
-    return response;
+- (BOOL)resignFirstResponder {
+
+    // Invalidate view layout.
+    // This can fail so we capture and act only when successful.
+    BOOL result = [super resignFirstResponder];
+    if (result == YES) {
```

Comment:
My thinking was to only invalidate the view layout if we are actually resigning first responder status. When logging from the code, I was seeing this being called multiple times. If those all fall within one pass of the run loop, it would probably only result in one view update but I wasn't sure about that.

Reply:
Any thoughts on that? Is it worth keeping?

I'd lean towards yes as it's a bit more explicit about the desired behaviour but it's a minor optimization at best.

Reply:
Honestly, the change makes a lot of sense. **However** it's possible we're inadvertently introducing UX glitches.

There's an old saying that goes like this.. ```premature optimization is the source of all evil```.

Personally, and by experience, I'd probably avoid changing the original behavior, **unless** we can determine exactly why it was introduced in the first place (OR) alternatively we detail a good Testing Plan, to make absolutely sure nothing breaks.

**Both alternatives are super.**

There are many hacks in this specific class, that, at a glance, don't really make any sense. For instance, the `editable = YES / NO` snippets are related to the text linkification. Editable TextViews don't process Links, and don't react to taps events, whereas Non Editable TextViews work pretty much like a WebView.

Reply:
Sounds good Jorge. I'll revert this to the original behaviour and add a comment based on the discussion here.

And thanks for clarifying the editable hack. There is a comment about data detectors but I wasn't sure what that meant. Good to know how it relates to link handling.